File: android/src/main/java/com/jhomlala/better_player/BetterPlayerCache.java
package com.jhomlala.better_player;

import android.content.Context;

import com.google.android.exoplayer2.database.ExoDatabaseProvider;
import com.google.android.exoplayer2.upstream.cache.LeastRecentlyUsedCacheEvictor;
import com.google.android.exoplayer2.upstream.cache.NoOpCacheEvictor;
import com.google.android.exoplayer2.upstream.cache.SimpleCache;

import java.io.File;

public class BetterPlayerCache {
    private static SimpleCache sDownloadCache;

    public static void createCache(Context context) {
        if (sDownloadCache == null) {
            sDownloadCache = new SimpleCache(
                    new File(context.getCacheDir(), "betterPlayerCache"),
                    new LeastRecentlyUsedCacheEvictor(100 * 1024 * 1024),
                    new ExoDatabaseProvider(context));
        }
    }

    public static SimpleCache getInstance() {
        return sDownloadCache;
    }

    public static void releaseCache() {
        sDownloadCache.release();
        sDownloadCache = null;
    }
}

File: android/src/main/java/com/jhomlala/better_player/BetterPlayerCache.kt
package com.jhomlala.better_player

import android.content.Context
import android.util.Log
import com.google.android.exoplayer2.upstream.cache.SimpleCache
import com.google.android.exoplayer2.upstream.cache.LeastRecentlyUsedCacheEvictor
import com.google.android.exoplayer2.database.ExoDatabaseProvider
import java.io.File
import java.lang.Exception

object BetterPlayerCache {
    @Volatile
    private var instance: SimpleCache? = null
    fun createCache(context: Context, cacheFileSize: Long): SimpleCache? {
        if (instance == null) {
            synchronized(BetterPlayerCache::class.java) {
                if (instance == null) {
                    instance = SimpleCache(
                        File(context.cacheDir, "betterPlayerCache"),
                        LeastRecentlyUsedCacheEvictor(cacheFileSize),
                        ExoDatabaseProvider(context)
                    )
                }
            }
        }
        return instance
    }

    @JvmStatic
    fun releaseCache() {
        try {
            if (instance != null) {
                instance!!.release()
                instance = null
            }
        } catch (exception: Exception) {
            Log.e("BetterPlayerCache", exception.toString())
        }
    }
}